### PR TITLE
docs/managed_vms: add endpoints sample

### PR DESCRIPTION
**Pull request by broady**

### CODE REVIEW

File: docs/managed_vms/endpoints/app.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package main
```
Comment (adg) [inline docs/managed_vms/endpoints/app.go]: copyright
doc comment

File: docs/managed_vms/endpoints/app.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	r.Path("/auth/info/googlejwt").Methods("GET").
+		HandlerFunc(authInfoHandler)
+	r.Path("/auth/info/googleidtoken").Methods("GET").
+		HandlerFunc(authInfoHandler)
+
+	http.Handle("/", r)
+	appengine.Main()
+}
+
+func echoHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (adg) [inline docs/managed_vms/endpoints/app.go]: doc comments please

File: docs/managed_vms/endpoints/client/main.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
+package main
```
Comment (adg) [inline docs/managed_vms/endpoints/client/main.go]: docs

File: docs/managed_vms/endpoints/client/main.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	var resp *http.Response
+	var err error
+	if *serviceAccount != "" {
+		resp, err = doJWT()
+	} else {
+		log.Fatal("Not implemented.")
```
Comment (adg) [inline docs/managed_vms/endpoints/client/main.go]: not implemented?

File: docs/managed_vms/endpoints/client/main.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	os.Stdout.Write(b)
+}
+
+func doJWT() (*http.Response, error) {
+	sa, err := ioutil.ReadFile(*serviceAccount)
+	if err != nil {
+		log.Fatalf("Could not read service account file: %v", err)
```
Comment (adg) [inline docs/managed_vms/endpoints/client/main.go]: return fmt.Errorf ?

File: docs/managed_vms/endpoints/client/main.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		Algorithm: "RS256",
+		Typ:       "JWT",
+	}
+
+	msg, err := jws.Encode(jwsHeader, jwt, rsaKey)
+	if err != nil {
+		log.Fatalf("Could not encode JWT: %v", err)
+	}
+
+	req, _ := http.NewRequest("GET", *host+"/auth/info/googlejwt?key="+*apiKey, nil)
```
Comment (adg) [inline docs/managed_vms/endpoints/client/main.go]: check the error?

File: docs/managed_vms/endpoints/client/main.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	return http.DefaultClient.Do(req)
+}
+
+// parseKey converts the binary contents of a private key file
+// to an *rsa.PrivateKey. It detects whether the private key is in a
+// PEM container or not. If so, it extracts the the private key
+// from PEM container before conversion. It only supports PEM
+// containers with no passphrase.
+func parseKey(key []byte) (*rsa.PrivateKey, error) {
+	block, _ := pem.Decode(key)
```
Comment (adg) [inline docs/managed_vms/endpoints/client/main.go]: check the error

File: docs/managed_vms/endpoints/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,106 @@
+swagger: "2.0"
```
Comment (adg) [inline docs/managed_vms/endpoints/swagger.yaml]: is this real life?

File: docs/managed_vms/endpoints/client/main.go (hunk 7 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: Woah. I actually copied this from the oauth library. We should fix that.

File: docs/managed_vms/endpoints/client/main.go (hunk 6 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: How could there be an error? I suppose if they put in a bad key.

File: docs/managed_vms/endpoints/client/main.go (hunk 7 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: Looks like it's intentional.

File: docs/managed_vms/endpoints/app.go (hunk 1 above)
Comment (broady) [inline docs/managed_vms/endpoints/app.go]: Done

File: docs/managed_vms/endpoints/client/main.go (hunk 4 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: Removed the id_token flow (need to think about how to do this from Go. We don't expose the id_token field in the oauth library).

File: docs/managed_vms/endpoints/client/main.go (hunk 5 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: Done for all

File: docs/managed_vms/endpoints/client/main.go (hunk 7 above)
Comment (broady) [inline docs/managed_vms/endpoints/client/main.go]: Added copyright notice. PTAL.

File: docs/managed_vms/endpoints/app.go [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		errorf(w, http.StatusInternalServerError, "Could not decode auth info: %v", err)
+		return
+	}
+	w.Write(b)
+}
+
+// errorf writes a swagger-compliant error response.
+func errorf(w http.ResponseWriter, code int, format string, a ...interface{}) {
+	var error struct {
```
Comment (adg) [inline docs/managed_vms/endpoints/app.go]: don't name this error, you're shadowing the built in type

Comment (broady) [inline docs/managed_vms/endpoints/app.go]: done, couldn't think of a better name than "out" though.

**Comment by adg** [comment]
LGTM

**Pull request merged**